src/main/kotlin/org/rust/lang/core/macros/RangeMap.kt

```diff
@@ -122,3 +123,20 @@ private fun MappedTextRange.dstIntersection(range: TextRange): MappedTextRange?
         null
     }
 }
+
+fun MutableList<MappedTextRange>.mergeAdd(range: MappedTextRange) {
```

Comment:
Actually, I don't fully understand why we need this. Could you explain, please (probably, it's worth adding a doc comment with an example)?

Reply:
It's an optimization to reduce the list size. The method replaces the last list element with a united range if they intersect